### PR TITLE
Update install-prereqs-arch script

### PR DESCRIPTION
**Pull request by patrickelectric**

### CODE REVIEW

**Comment by rmackay9** [comment]
This looks good and harmless to me.

**Comment by OXINARF** [comment]
@patrickelectric Second commit seems fine, I don't quite understand the first one. What is the problem with running it inside the ArduPilot path?

**Comment by patrickelectric** [comment]
@OXINARF 
https://github.com/patrickelectric/ardupilot/blob/arch/Tools/scripts/install-prereqs-arch.sh#L91
```sh
(
 cd ./ardupilot
 git submodule init
 git submodule update
)
```

**Comment by lucasdemarchi** [comment]
@patrickelectric that's the line that need to be changed then. We can assume the script is inside a git repository, so in that lines we do:

```bash
SCRIPT_DIR=$(dirname $(realpath ${BASH_SOURCE[0]}))
(
    cd $SCRIPT_DIR
    git submodule init
    git submodule update --recursive
)
```

Note there's a `--recursive` missing there as well.

File: Tools/scripts/install-prereqs-arch.sh [hunk 1]
```diff
@@ -4,6 +4,14 @@ set -x
 
 command -v yaourt >/dev/null 2>&1 || { echo >&2 "Please install yaourt first. Aborting."; exit 1; }
 
+SCRIPT_NAME=`basename "$0"`
+
+if [ ! -d "ardupilot" ]; then
+  echo >&2 "Please run script out of ardupilot directory."
+  echo >&2 "E.g: ./ardupilot/Tools/scripts/$SCRIPT_NAME"
+  exit 1
```
Comment (lucasdemarchi) [inline Tools/scripts/install-prereqs-arch.sh]: indent for bash scripts is 4 spaces

File: Tools/scripts/install-prereqs-arch.sh [hunk 2]
```diff
@@ -42,8 +50,12 @@ yaourt -S --noconfirm $ARCH_AUR_PKGS
 
 (
  cd /usr/lib/ccache
- sudo ln -s /usr/bin/ccache arm-none-eabi-g++
- sudo ln -s /usr/bin/ccache arm-none-eabi-gcc
+ if [ ! -f arm-none-eabi-g++ ]; then
+  sudo ln -s /usr/bin/ccache arm-none-eabi-g++
```
Comment (lucasdemarchi) [inline Tools/scripts/install-prereqs-arch.sh]: ditto

**Comment by patrickelectric** [comment]
Done.

**Comment by rmackay9** [comment]
This is a little outside my area but I'm going to push this in in any case.  If it's wrong I suspect others will cry soon and it shouldn't lead to vehicles crashing.

Thanks @patrickelectric!

**Pull request merged**